### PR TITLE
Fix boto3 util pack error

### PR DESCRIPTION
**Pull request by aneeshusa**
### What does this PR do?

Removes a ton of logspam that I inadvertently introduced in #42624.

The first commit removes the only usage of `__salt__` in the boto3 utils module.
The second commit removes the bad usage of the `pack` argument from `boto_sqs`.
This is only available for the boto utils module, not in boto3.
The boto utils module needs this as it uses `__salt__` internally,
but boto3 does not and thus does not need a `pack` argument.

### What issues does this PR fix or reference?

#30279 - First addition of `pack` to boto utils module.
#30867 - Usage of `pack` to all boto* modules.
#30041 - Added unused `boto3.get_role_arn` function
#42624 - Switch of boto_sqs to boto3.

### Previous Behavior
Tons of logspam in Jenkins output.

### New Behavior
Logspam is gone.

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by aneeshusa** [comment]
cc @kraney Do you know why you added the `boto3.get_role_arn` function in #30041?

**Comment by rallytime** [comment]
@aneeshusa It looks to me like that was added to perhaps replace the uses of the private functions in some of the boto_* modules/states that are defining a `_get_role_arn`, but the references were never updated. I checked some of these private functions against this one (but not all) and they're all quite similar, if not exactly the same.

It might be useful to keep that function, but then use it for DRY. Thoughts?

File: salt/modules/boto_sqs.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def __virtual__():
     '''
     if not HAS_BOTO3:
         return (False, 'The boto_sqs module could not be loaded: boto3 libraries not found')
-    __utils__['boto3.assign_funcs'](__name__, 'sqs', pack=__salt__)
+    __utils__['boto3.assign_funcs'](__name__, 'sqs')
```
Comment (rallytime) [inline salt/modules/boto_sqs.py]: 🎉

**Comment by aneeshusa** [comment]
@rallytime I think it would make more sense to put that function into the `boto_iam` execution module. That way we can have DRY but avoid needing to have `__salt__` in the `boto3.py` utils file. I've never liked the `pack` workaround and I'd like to avoid it entirely for `boto3`, which is why I'm proposing this approach.

Long term I'd also like to see `pack` removed from the `boto.py` utils file - we can take the same approach as `boto3.py` and make a copy of the `config.option` function locally to avoid the dependency. Or, we could make `config.option` available from a `utils` module and use it from there.

Let me know if this sounds good to you.

**Comment by rallytime** [comment]
@aneeshusa That sounds great 👍

**Comment by aneeshusa** [comment]
@rallytime OK, mind if I save those changes (moving `get_role_arn` into `boto_iam`, and removing `pack`) for followups? I want to get rid of the logspam as soon as possible.

**Comment by cachedout** [comment]
Just making sure that all of @saltstack/team-boto sees this before we merge it.

**Comment by rallytime** [comment]
@aneeshusa Yep, sounds good. I restarted the tests because the results have timed out.

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
Test failures are unrelated.

**Pull request merged**